text: add option to change pressed tab into spaces

judaszszymon opened the pull request:
Add option in models/prefs.py and widgets/prefs.py to override tab with spaces
Override keyPressEvent in TextEdit with new behaviour (intercept tab key when
option is checked and post new keypress and keyrelease events with spaces).

Reported-by: @conz27 #660
Signed-off-by: Szymon Judasz szymon.judasz@student.uj.edu.pl

davvid commented on cola/widgets/text.py:
```diff
@@ -292,6 +291,28 @@ def wheelEvent(self, event):
+            new_release_event = QtGui.QKeyEvent(QtGui.QKeyEvent.KeyRelease, Qt.Key_A, Qt.NoModifier, tab_width * ' ')
+            event.accept()
+            QtCore.QCoreApplication.postEvent(self, new_press_event)
+            QtCore.QCoreApplication.postEvent(self, new_release_event)
```
Hmm I wonder if it'd be simpler to get the text cursor and insert the spaces instead of creating new events that insert spaces.

It's also kinda funky and probably not advised to send a "Key" press event that sends a non-key (aka multiple spaces).  I'm surprised that works.

Instead of posting new events, grab the textcursor and insert spaces.  That should be more robust.

davvid commented on cola/models/prefs.py:
```diff
@@ -90,6 +91,9 @@ def linebreak():
 
+def override_tab_with_spaces():
+    return gitcfg.current().get(OVERRIDE_TAB_WITH_SPACES, False)
+
```
Sorry to bikeshed, but please follow vim's naming convention for this feature.

We should call it `cola.expandtab`.  The description would be more precise if it said, "Insert spaces instead of tabs" instead of "Override ...".

judaszszymon replied:
Okay, I'll change the code soon.

judaszszymon commented:
@davvid I addressed your comments:
* Changed inserting spaces method to grabbing cursor
* new name for configuration variable `cola.expandtab` and description in .rst file
Sorry that it took so long.
Can you review again?
